### PR TITLE
UCT/API: remove status arg from uct_completion_callback_t

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
remove status arg from uct_completion_callback_t

## Why ?
uct_completion_t contains actual status
https://github.com/openucx/ucx/pull/5631#discussion_r481919538

### CODE REVIEW

File: src/ucp/core/ucp_am.c [hunk 1]
```diff
@@ -527,10 +527,11 @@ static void ucp_am_zcopy_req_complete(ucp_request_t *req, ucs_status_t status)
...
 }
 
-void ucp_am_zcopy_completion(uct_completion_t *self, ucs_status_t status)
+void ucp_am_zcopy_completion(uct_completion_t *self)
 {
-    ucp_request_t *req = ucs_container_of(self, ucp_request_t,
-                                          send.state.uct_comp);
+    ucp_request_t *req  = ucs_container_of(self, ucp_request_t,
+                                           send.state.uct_comp);
+    ucs_status_t status = self->status;
```
Comment (yosefe) [inline src/ucp/core/ucp_am.c:534]: maybe remove local var status

File: src/ucp/core/ucp_request.inl [hunk 2]
```diff
@@ -689,4 +689,12 @@ ucp_send_request_get_id(ucp_request_t *req)
                                      ucp_ep_use_indirect_id(req->send.ep));
 }
 
+static UCS_F_ALWAYS_INLINE
+void ucp_completion_update_status(uct_completion_t *comp, ucs_status_t status)
+{
+    if (ucs_unlikely(status != UCS_OK) && (comp->status == UCS_OK)) {
+        comp->status = status;
+    }
+}
```
Comment (yosefe) [inline src/ucp/core/ucp_request.inl:698]: maybe put this in UCT API? seems it's common for both UCT and UCP

File: src/ucp/proto/proto_am.c [hunk 3]
```diff
@@ -92,11 +92,12 @@ void ucp_proto_am_zcopy_req_complete(ucp_request_t *req, ucs_status_t status)
...
 
-void ucp_proto_am_zcopy_completion(uct_completion_t *self,
-                                    ucs_status_t status)
+void ucp_proto_am_zcopy_completion(uct_completion_t *self)
 {
-    ucp_request_t *req = ucs_container_of(self, ucp_request_t,
-                                          send.state.uct_comp);
+    ucp_request_t *req  = ucs_container_of(self, ucp_request_t,
+                                           send.state.uct_comp);
+    ucs_status_t status = self->status;
```
Comment (yosefe) [inline src/ucp/proto/proto_am.c:99]: remove local var

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@shamisp @tonycurtis can you pls review UCT API part?
@evgeny-leksikov pls squash

**Comment by shamisp** [comment]
👍

**Review by tonycurtis — APPROVED** [review]

**Pull request merged**